### ISSUE TITLE
PRS & CT02 raised for valid databricks window functions

### ISSUE THREAD

**Issue by vvveracruz** [sqlfluff/sqlfluff#4851]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The following line raised a PRS & CT02 error:
```sql
col_1 OVER (ORDER BY col_2)
```
raised
```
L: 215 | P:  65 | CP02 | Unquoted identifiers must be consistently lower case.
                       | [capitalisation.identifiers]
L: 215 | P:  70 |  PRS | Line 213, Position 70: Found unparsable section: ' BY
                       | col_2
```

This was resolved when making the `ORDER` keyword lowercase (not desirable) or updating the window funtion to 
```
col_1 OVER (PARTITION BY TRUE ORDER BY col_2)
```

In all fairness the [databricks docs](https://docs.databricks.com/sql/language-manual/sql-ref-window-functions.html#syntax) do kind of make it seem like the `PARTITION BY` statement is required, but the first version runs fine.

### Expected Behaviour

No PRS or CT02 failure – fluff parses window statements without a partition by clause fine :)

### Observed Behaviour

(I think it's fully explained under `What Happened` but open to qs/amendments.)

### How to reproduce

`model.sql` file:
```sql
SELECT 
col_1 OVER (ORDER BY col_2)
FROM {{ ref('upstream_model') }}
```

commands:
`pipenv run sqlfluff lint model.sql`

### Dialect

databricks

### Version

- sqlfluff v2.0.0"
- sqlfluff-templater-dbt v2.0.0
- python_version v3.10
- dbt-core v1.4.1
- dbt-databricks v1.4.2

### Configuration

```
[sqlfluff]
verbose = 0
nocolor = False
dialect = databricks
templater = dbt
rules = None
exclude_rules = L031, L001, L008, ST06
recurse = 0
output_line_length = 80
runaway_limit = 10
ignore_templated_areas = True
encoding = autodetect
disable_noqa = False
# Comma separated list of file extensions to lint.
# NB: This config will only apply in the root folder.
sql_file_exts = .sql,.sql.j2,.dml,.ddl
max_line_length = 99

[sqlfluff:indentation]
indent_unit = space
tab_space_size = 2
indented_joins = False
indented_using_on = True
template_blocks_indent = True

[sqlfluff:layout:type:binary_operator] # Keywords - migrated from L007
line_position = leading

[sqlfluff:layout:type:comma] # Migrated from L019
line_position = leading

[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:dbt]
...

[sqlfluff:templater:dbt:context]
...

[sqlfluff:templater:dbt:macros]
...

# Some rules can be configured directly from the config common to other rules.
[sqlfluff:rules]
allow_scalar = True
single_table_references = consistent
unquoted_identifiers_policy = all
indented_on_contents = False

# Some rules have their own specific config.

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper
extended_capitalisation_policy = upper

[sqlfluff:rules:aliasing.table]
aliasing = explicit

[sqlfluff:rules:aliasing.column]
aliasing = explicit

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = consistent

[sqlfluff:rules:layout.long_lines]
ignore_comment_lines = False

[sqlfluff:rules:references.from]
force_enable = False

[sqlfluff:rules:references.consistent]
force_enable = False

[sqlfluff:rules:references.keywords]
unquoted_identifiers_policy = aliases
quoted_identifiers_policy = none

[sqlfluff:rules:capitalisation.functions]  # Function names
capitalisation_policy = upper
extended_capitalisation_policy = upper

[sqlfluff:rules:convention.select_trailing_comma]
select_clause_trailing_comma = forbid

[sqlfluff:rules:capitalisation.literals]  # Null & Boolean Literals
capitalisation_policy = upper

[sqlfluff:rules:structure.subquery]
# By default, allow subqueries in from clauses, but not join clauses.
forbid_subquery_in = both

[sqlfluff:rules:convention.count_rows]  # Consistent syntax to count all rows
prefer_count_1 = False
prefer_count_0 = False

[sqlfluff:rules:convention.terminator]  # Semi-colon formatting approach.
multiline_newline = False
require_final_semicolon = False

[sqlfluff:rules:ambiguous.column_references]  # GROUP BY/ORDER BY column references.
group_by_and_order_by_style = explicit

[sqlfluff:rules:references.special_chars]  # Special characters in identifiers
unquoted_identifiers_policy = all
quoted_identifiers_policy = all
allow_space_in_identifier = False
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
